File: java-impl/src/UndirectedGraph.java
import java.util.ArrayList;

public class UndirectedGraph implements Graph {
 Integer nodes = 0;
 ArrayList<ArrayList<Integer>> graph;

 public UndirectedGraph(Integer nodes) {
  this.nodes = nodes;
  this.graph = new ArrayList<ArrayList<Integer>>();
  for (int i = 0; i < this.nodes; i ++)
   this.graph.add(i, new ArrayList<Integer>());
 }
 
 public void addEdge(Integer source, Integer destination) {
  graph.get(source).add(destination);
  graph.get(destination).add(source);
 }

 public void printGraph() {
  for (int i = 0; i < nodes; i++) {
   if (graph.get(i).size() > 0) {
    System.out.print(i + " => ");
    for (int j = 0; j < graph.get(i).size(); j ++) 
     System.out.print(graph.get(i).get(j) + " ");
    System.out.println();
   }
  }
 }
}

File: typescript-impl/src/UndirectedGraph.ts
import { Graph } from "./Graph";

export class UndirectedGraph implements Graph {
 nodes: number = 0;
 graph: Array<Array<number>>;
 
 constructor(nodes: number) {
  this.nodes = nodes;
  this.graph = [];
  for (let i = 0; i < this.nodes; i++)
   this.graph[i] = [];
 }
 
 public addEdge(source: number, destination: number): void {
  this.graph[source].push(destination);
  this.graph[destination].push(source);
 }

 public printGraph(): void {
  for (let i = 0; i < this.nodes; i++) {
   if (this.graph[i].length > 0) {
    for (let j = 0; j < this.graph[i].length; j++) {
     console.log(i, "=>", this.graph[i][j]);
    }
   }
  }
 }
}
